File: python/0118杨辉三角.py

def generate(numRows: int) -> list:
    
    res=[[1],[1,1]]
    if(numRows<2):
        return res[:numRows]

    numRows-=1
    for n in range(2,numRows+1):
        crt=[1 if i==0 or i==n else res[n-1][i-1]+res[n-1][i] for i in range(n+1)]
        res.append(crt)
    
    return res

print(generate(15))

File: csharp/0118杨辉三角.cs
public partial class Solution
{
    public IList<IList<int>> Generate(int numRows)
    {
        IList<IList<int>> res = new List<IList<int>>(){
            new List<int>(){-1},
            new List<int>(){1},
            new List<int>(){1,1},
        };

        if(numRows>2)
        {
            for (int i = 3; i<=numRows;i++)
            {
                IList<int> crt = new List<int>() { 1 };

                for (int j = 1; j < i-1; j++)
                {
                    crt.Add(res[i-1][j-1]+res[i-1][j]);
                }
                crt.Add(1);
                res.Add(crt);
            }
        }

        return res.Take(new Range(1, numRows+1)).ToList();
    }
}